### PR TITLE
Fix incorrect superuser time constraint warning logging

### PR DESCRIPTION
**Pull request by danielgustafsson**
The check for role time constraints had a bug in the superuser check such that it would raise a WARNING every time a superuser role was checked regardless of if it had constraints added against it. This was causing log-spamming of erroneous entries.

Fix by only raising a WARNING in case there were any constraints added.

This needs to be backported to 6X as well.

### CODE REVIEW

**Comment by danielgustafsson** [comment]
@ashwinstar does this make the warning you spotted go away for you?

File: src/backend/libpq/auth.c [hunk 1]
```diff
@@ -3158,6 +3158,8 @@ check_auth_time_constraints_internal(char *rolname, TimestampTz timestamp)
 	HeapTuple		tuple;
 	authPoint 		now;
 	int				status;
+	bool			isSuper;
```
Comment (ashwinstar) [inline src/backend/libpq/auth.c]: `isSuper` reads little short hand, can we make it `isRoleSuperuser`

Comment (danielgustafsson) [reply src/backend/libpq/auth.c]: Sure, I'll update to that before pushing.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks for review!

**Pull request closed without merging**